de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs

```diff
@@ -90,8 +92,20 @@ public class StringDecrypterInfo {
 					var ldci4 = instrs[i + 3];
 					if (!ldci4.IsLdcI4())
 						continue;
-
-					var info = new StringDecrypterInfo(method, ldci4.GetLdcI4Value());
+					var ldarg1 = instrs[i + 4];
+					if (!ldarg1.IsLdarg() || ldarg1.GetParameterIndex() != 1)
+						continue;
+					var opAdd1 = instrs[i + 5];
+					int magicAddInt = 0;
+					bool newVersion = false;
+					if (opAdd1.OpCode == OpCodes.Add) {
+						var ldci4_2 = instrs[i + 6];
```

Comment:
You haven't verified that i + 6 is a valid index here or in the other new code above.

Reply:
ok